File: OdonSys-SPA/src/app/core/models/view/client-model.ts
import { Country } from "../../enums/country.enum";
import { DoctorApiModel } from "../api/doctor/doctor-api-model";

export class ClientModel {
  constructor(
    public id: string,
    public active: boolean,
    public dateCreated: Date,
    public dateModified: Date,
    public name: string,
    public middleName: string,
    public surname: string,
    public secondSurname: string,
    public document: string,
    public ruc: string,
    public country: Country,
    public debts: boolean,
    public phone: string,
    public email: string,
    public doctors: DoctorApiModel[]
  ) {}
}

File: OdonSysBackEnd/Resource/Contract/Contract.Administration/Clients/ClientModel.cs
using Contract.Administration.Users;
using Utilities.Enums;

namespace Contract.Administration.Clients;

public class ClientModel
{
    public string Id { get; set; }
    public bool Active { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime DateModified { get; set; }
    public string Name { get; set; }
    public string MiddleName { get; set; }
    public string Surname { get; set; }
    public string SecondSurname { get; set; }
    public string Document { get; set; }
    public string Ruc { get; set; }
    public Country Country { get; set; }
    public bool Debts { get; set; }
    public string Phone { get; set; }
    public string Email { get; set; }
    public IEnumerable<DoctorModel> Doctors { get; set; }
}
